docs/site/migration/auth/built-in.md

```diff
@@ -1,17 +1,290 @@
+? Allow additional (free-form) properties? Yes
+Model UserCredentials will be created in src/models/user-credentials.model.ts
+
+Lets add a property to UserCredentials
+Enter an empty property name when done
+
+? Enter the property name: id
+? Property type: number
+? Is id the ID property? Yes
+? Is id generated automatically? No
+? Is it required?: Yes
+? Default value [leave blank for none]:
+
+Lets add another property to UserCredentials
+Enter an empty property name when done
+
+? Enter the property name: password
+? Property type: string
+? Is it required?: Yes
+? Default value [leave blank for none]:
+
+Lets add another property to UserCredentials
+Enter an empty property name when done
+
+? Enter the property name: userId
+? Property type: number
+? Is it required?: Yes
+? Default value [leave blank for none]:
+
+Lets add another property to UserCredentials
+Enter an empty property name when done
+
+? Enter the property name:
+   create src/models/user-credentials.model.ts
+   update src/models/index.ts
+
+Model UserCredentials was created in src/models/
+```
+
+### User Persistency and CRUD APIs
+
+First let's create a memory datasource that inherits from LoopBack 3 for data
+storage.
+
+```sh
+$ lb4 datasource
+? Datasource name: db
+? Select the connector for mysqlDs: In-memory db (supported by StrongLoop)
+? window.localStorage key to use for persistence (browser only):
+? Full path to file for persistence (server only):
+```
+
+Then create the corresponding repositories for `User` and `UserCredentials`:
+
+```sh
+cd myLB4App
+$ lb4 repository
+? Please select the datasource DbDatasource
+? Select the model(s) you want to generate a repository User, UserCredentials
+? Please select the repository base class DefaultCrudRepository (Legacy juggler bridge)
+```
+
+Create the user controller for CRUD operations:
+
+```sh
+$ lb4 controller
+? Controller class name: User
+Controller User will be created in src/controllers/user.controller.ts
+
+? What kind of controller would you like to generate? REST Controller with CRUD functions
+? What is the name of the model to use with this CRUD repository? User
+? What is the name of your CRUD repository? UserRepository
+? What is the name of ID property? id
+? What is the type of your ID? number
+? Is the id omitted when creating a new instance? No
+? What is the base HTTP path name of the CRUD operations? /users
+   create src/controllers/user.controller.ts
+   update src/controllers/index.ts
+
+Controller User was created in src/controllers/
+```
+
+### Creating Login Function
+
+_Migrating all functions for User need more effort, this section only create
+login function as the minimum requirement._
+
+For token based authentication, a user logs in by providing correct credentials
+(email and password) in the payload of endpoint `User/login`, then gets a token
+back with its identity information encoded and includes it in the header of next
```

Comment:
maybe `and includes the token in the header`? (I found it a bit difficult to read with several pronouns in a sentence)